src/Microsoft.ML.StaticPipe/OnlineLearnerStatic.cs

```diff
@@ -64,7 +62,67 @@ public static class AveragedPerceptronStaticExtensions
+
+            return rec.Output;
+        }
+
+        /// <summary>
+        /// Predict a target using a linear binary classification model trained with the AveragedPerceptron trainer, and a custom loss.
+        /// </summary>
+        /// <param name="ctx">The binary classification context trainer object.</param>
+        /// <param name="label">The label, or dependent variable.</param>
+        /// <param name="features">The features, or independent variables.</param>
+        /// <param name="lossFunction">The custom loss.</param>
+        /// <param name="weights">The optional example weights.</param>
+        /// <param name="options">Advanced arguments to the algorithm.</param>
+        /// <param name="onFit">A delegate that is called every time the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}.Fit(DataView{TInShape})"/> method is called on the
+        /// <see cref="Estimator{TInShape, TOutShape, TTransformer}"/> instance created out of this. This delegate will receive
+        /// the linear model that was trained, as well as the calibrator on top of that model. Note that this action cannot change the
+        /// result in any way; it is only a way for the caller to be informed about what was learnt.</param>
+        /// <returns>The set of output columns including in order the predicted binary classification score (which will range
+        /// from negative to positive infinity), and the predicted label.</returns>
+        /// <seealso cref="AveragedPerceptronTrainer"/>.
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[AveragedPerceptron](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Static/AveragedPerceptronBinaryClassification.cs)]
+        /// ]]></format>
+        /// </example>
+        public static (Scalar<float> score, Scalar<bool> predictedLabel) AveragedPerceptron(
+                this BinaryClassificationContext.BinaryClassificationTrainers ctx,
+                Scalar<bool> label,
+                Vector<float> features,
+                Scalar<float> weights,
+                IClassificationLoss lossFunction,
+                AveragedPerceptronTrainer.Options options,
+                Action<LinearBinaryModelParameters> onFit = null
+            )
+        {
+            Contracts.CheckValue(label, nameof(label));
+            Contracts.CheckValue(features, nameof(features));
+            Contracts.CheckValueOrNull(weights);
+            Contracts.CheckValueOrNull(options);
+            Contracts.CheckValueOrNull(onFit);
+
+            bool hasProbs = lossFunction is LogLoss;
+
+            var rec = new TrainerEstimatorReconciler.BinaryClassifierNoCalibration(
+                (env, labelName, featuresName, weightsName) =>
+                {
+                    options.LabelColumn = labelName;
+                    options.FeatureColumn = featuresName;
+                    options.InitialWeights = weightsName;
```

Comment:
>InitialWeights [](start = 28, length = 14)

just checking that this is not an Optional<T>

Reply:
yeap . its of type `string` in  OnlineLinear.cs 
`public string InitialWeights;`

---
In reply to: [248544431](https://github.com/dotnet/machinelearning/pull/2163#discussion_r248544431) [](ancestors = 248544431)